cmd/eksctl-anywhere/cmd/downloadpackages.go

```diff
@@ -0,0 +1,122 @@
+	Long:         `Download all the EKS Anywhere curated package images and helm charts.`,
+	SilenceUsage: true,
+	RunE:         runDownloadPackages,
+	Args: func(cmd *cobra.Command, args []string) error {
+		if err := cobra.ExactArgs(1)(cmd, args); err != nil {
+			return fmt.Errorf("A destination directory must be specified as an argument")
+		}
+		return nil
+	},
+}
+
+func init() {
+	downloadCmd.AddCommand(downloadPackagesCmd)
+
+	downloadPackagesCmd.Flags().StringVarP(&downloadPackagesCommand.bundleFile, "bundle", "b", "", "EKS-A bundle file to read artifact dependencies from")
+	if err := downloadPackagesCmd.MarkFlagRequired("bundle"); err != nil {
+		log.Fatalf("Cannot mark 'bundle' flag as required: %s", err)
+	}
+	downloadPackagesCmd.Flags().StringVarP(&downloadPackagesCommand.awsRegion, "aws-region", "", os.Getenv(config.EksaRegionEnv), "Region to download images from")
+}
+
+var downloadPackagesCommand = DownloadPackagesCommand{}
+
+// DownloadPackagesCommand copies packages specified in a bundle to a destination.
+type DownloadPackagesCommand struct {
+	destination   string
+	bundleFile    string
+	awsRegion     string
+	registryCache *registry.Cache
+}
+
+func runDownloadPackages(_ *cobra.Command, args []string) error {
+	ctx := context.Background()
+	downloadPackagesCommand.destination = args[0]
+
+	credentialStore := registry.NewCredentialStore()
+	err := credentialStore.Init()
+	if err != nil {
+		return err
+	}
+
+	return downloadPackagesCommand.call(ctx, credentialStore)
+}
+
+func (c DownloadPackagesCommand) call(ctx context.Context, credentialStore *registry.CredentialStore) error {
+	factory := dependencies.NewFactory()
+	deps, err := factory.
+		WithManifestReader().
+		Build(ctx)
+	if err != nil {
+		return err
+	}
+
+	eksaBundle, err := bundles.Read(deps.ManifestReader, c.bundleFile)
+	if err != nil {
+		return err
+	}
+
+	c.registryCache = registry.NewCache()
+	bundleReader := curatedpackages.NewPackageReader(c.registryCache, credentialStore, c.awsRegion)
+
+	imageList := bundleReader.ReadChartsFromBundles(ctx, eksaBundle)
+
+	log.Printf("Downloading curated packages helm charts from public ECR to %s", c.destination)
+	err = c.downloadImages(imageList, c.destination, "")
+	if err != nil {
+		return err
+	}
+
+	imageList, err = bundleReader.ReadImagesFromBundles(ctx, eksaBundle)
+	if err != nil {
+		return err
+	}
+	log.Printf("Downloading curated packages images from private ECR to %s", c.destination)
+	return c.downloadImages(imageList, c.destination, "curated-packages")
+}
+
+func (c DownloadPackagesCommand) downloadImages(imageList []registry.Artifact, destination, project string) error {
+	for _, image := range imageList {
+		source := image.VersionedImage()
+		destination = path.Join(destination, project, image.Repository)
+		err := downloadSingleImage(source, destination)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func downloadSingleImage(source, destination string) error {
```

Comment:
if we could make this reusable by moving it somewhere in pkg that would be great